packages/repack/src/modules/ScriptManager/__tests__/ScriptManager.test.ts

```diff
@@ -2,7 +2,13 @@ import { Script } from '../Script';
 import { ScriptManager } from '../ScriptManager';
 
 jest.mock('../NativeScriptManager', () => ({
-  loadScript: jest.fn(),
+  loadScript: jest.fn((locator) =>
+    locator.fetch
+      ? new Promise<void>((resolve) => {
+          setTimeout(() => resolve(), 10);
+        })
+      : Promise.resolve()
+  ),
```

Comment:
can we only apply this mock implementation to the selected tests that we need? I think this can be achieved with `jest.spyOn`